packages/components/pie-modal/test/component/pie-modal.spec.ts

```diff
@@ -1,31 +1,47 @@
 import { test, expect } from '@sand4rt/experimental-ct-web';
 import { type Page } from '@playwright/test';
+import { PieButton } from '@justeattakeaway/pie-button';
 import { PieIconButton } from '@justeattakeaway/pie-icon-button';
 import {
     WebComponentTestWrapper,
 } from '@justeattakeaway/pie-webc-testing/src/helpers/components/web-component-test-wrapper/WebComponentTestWrapper.ts';
 import { renderTestPieModal } from '../helpers/index.ts';
 
 import { PieModal } from '@/index';
-import { headingLevels, ON_MODAL_BACK_EVENT } from '@/defs';
+import {
+    ON_MODAL_BACK_EVENT,
+    ON_MODAL_CLOSE_EVENT,
+    headingLevels,
+} from '@/defs';
 
-const closeButtonSelector = '[data-test-id="modal-close-button"]';
+const modalSelector = '[data-test-id="pie-modal"]';
 const backButtonSelector = '[data-test-id="modal-back-button"]';
+const closeButtonSelector = '[data-test-id="modal-close-button"]';
 
-// Mount any components that are used inside of pie-modal so that
+// Mount then unmount any components that are used inside of pie-modal so that
 // they have been registered with the browser before the tests run.
 // There is likely a nicer way to do this but this will temporarily
 // unblock tests.
-test.beforeEach(async ({ page, mount }) => {
-    await mount(
-        PieIconButton,
-        {},
-    );
-
-    // Removing the element so it's not present in the tests (but is still registered in the DOM)
-    await page.evaluate(() => {
-        const element : Element | null = document.querySelector('pie-icon-button');
-        element?.remove();
+test.beforeEach(async ({ mount }) => {
+    await (await mount(PieButton)).unmount();
+    await (await mount(PieIconButton)).unmount();
+});
```

Comment:
I have absolutely no idea why `unmount` didn't work for me when I initial wrote this!